src/common/options/global.yaml.in

```diff
@@ -4000,15 +4000,39 @@ options:
   default: false
   with_legacy: true
 - name: bdev_enable_discard
+  desc: send discards to the block device
   type: bool
   level: advanced
   default: false
   with_legacy: true
+  flags:
+  - runtime
+  see_also:
+  - bdev_async_discard
+  - bdev_async_discard_threads
 - name: bdev_async_discard
+  desc: send discards to the block device in one or more threads
   type: bool
   level: advanced
   default: false
   with_legacy: true
+  flags:
+  - runtime
+  see_also:
+  - bdev_enable_discard
+  - bdev_async_discard_threads
+- name: bdev_async_discard_threads
+  desc: number of discard threads used to issue discards to the device
+  type: bool
+  level: advanced
+  default: 1
+  min: 1
+  with_legacy: true
```

Comment:
since you always access this option using the new-style `get_val<..>`, there is no need to use `with_legacy`.

Reply:
Thanks! I'm unfamiliar with the legacy flag on options, I suppose it's the difference between filling out a struct with options vs the templated one in use here. I'll take a look to see.